### PR TITLE
Error page: Use descriptive code for alloptions

### LINKED ISSUE #1210: Alloptions: Better error code when size exceeds limit
When the alloptions size exceeds the defined limit (1MB), we display an error page with a `1024` code. Let's make that code more human friendly so it's easier to identify the actual problem...`ALLOPTIONS_SIZE_LIMIT_EXCEEDED` or similar.

### PR DESCRIPTION
**Pull request by GaryJones**
## Description

When the alloptions size exceeds the defined limit (1MB), we displayed an error page with a `1024` code. We can make that code more human-friendly so it's easier to identify the actual problem.

Fixes #1210

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

No real testing needed (unless there are e2e tests hiding somewhere) - it's just a change in the visible error code on an error page.

### CODE REVIEW

**Comment by mjangda** [comment]
Background on the error code: we intentionally left it not-obvious, since end users are the ones primarily seeing the error page. An error like `ALLOPTIONS_SIZE_LIMIT_EXCEEDED` can be panic inducing especially considering they have no control over it.

There's probably a middle ground we can find like `Error 1024 (alloptions)` instead?

File: errors/alloptions-limit.html [hunk 1]
```diff
@@ -146,7 +146,7 @@
 				<h1>Site Unavailable</h1>
 				<p>This site is temporarily unavailable, please try again in a few moments.</p>
 				<footer>
-					<span class="error">Error 1024</span>
+					<span class="error">Error <code>ALLOPTIONS_SIZE_LIMIT_EXCEEDED</code></span>
```
Comment (GaryJones) [inline errors/alloptions-limit.html]:
```suggestion
					<span class="error">Error 1024 (<code>alloptions</code>)</span>
```

**Comment by GaryJones** [comment]
Background makes sense - message updated.

**Comment by mjangda** [comment]
r858-stacks

**Pull request merged**